L/LLVM/common.jl

```diff
@@ -451,7 +451,7 @@ function configure_build(ARGS, version; experimental_platforms=false, assert=fal
         LibraryProduct("libclang", :libclang, dont_dlopen=true),
         LibraryProduct(["LTO", "libLTO"], :liblto, dont_dlopen=true),
         ExecutableProduct("llvm-config", :llvm_config, "tools"),
-        ExecutableProduct(["clang", "clang-$(version.major)"], :clang, "tools"),
+        ExecutableProduct(["clang", "clang-$(version.major)"], :clang, "bin"),
```

Comment:
Gate this based on LLVM version.

Reply:
Added option, but based on whether we are building ROCmLLVM or not...